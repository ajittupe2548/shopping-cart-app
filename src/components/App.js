import React from "react";
import Header from "./Header";
import ProductList from "./ProductList";
import Cart from "./Cart";

function App() {
  return (
    <>
      <Header />
      <ProductList />
      <Cart />
    </>
  );
}

export default App;
