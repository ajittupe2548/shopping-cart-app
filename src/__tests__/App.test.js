import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import App from "../components/App";
import productsReducer from "../store/productsSlice";
import cartReducer from "../store/cartSlice";

// Mock child components to isolate App component testing
jest.mock("../components/Header", () => {
  return function MockHeader() {
    return <div data-testid="header">Header</div>;
  };
});

jest.mock("../components/ProductList", () => {
  return function MockProductList() {
    return <div data-testid="product-list">ProductList</div>;
  };
});

jest.mock("../components/Cart", () => {
  return function MockCart() {
    return <div data-testid="cart">Cart</div>;
  };
});

const createTestStore = () => {
  return configureStore({
    reducer: {
      products: productsReducer,
      cart: cartReducer,
    },
  });
};

describe("App Component", () => {
  let store;

  beforeEach(() => {
    store = createTestStore();
  });

  test("renders without crashing", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  test("renders all main components", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(getByTestId("header")).toBeInTheDocument();
    expect(getByTestId("product-list")).toBeInTheDocument();
    expect(getByTestId("cart")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
