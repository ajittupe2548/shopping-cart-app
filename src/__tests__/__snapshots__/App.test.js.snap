// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App Component matches snapshot 1`] = `
<div
  data-testid="header"
>
  Header
</div>
`;
